sklearn/preprocessing/_discretization.py

```diff
@@ -279,9 +292,9 @@ def transform(self, X):
         np.clip(Xt, 0, self.n_bins_ - 1, out=Xt)
 
         if self.encode == 'ordinal':
-            return Xt
+            return Xt.astype(self.dtype)
```

Comment:
I'm not sure it makes sense to do this for ordinal. Let's wait for a second option about it.